distribution/std-lib/Standard/src/Base/Data/Any/Extensions.enso

```diff
@@ -45,61 +64,117 @@ Any.!= that = (this == that).not
 
 ## Checks if `this` is greater than `that`.
 
+   Arguments:
+   - that: The value to compare `this` against.
+
    To have `>` defined, a type must define `compare_to`, returning an Ordering.
 
+   ! Implementing Greater Than
+     Many types can admit a definition of greater than that is more efficient
+     than the generic one given here. When implementing this for your own types
+     please ensure that it is semantically equivalent to this definition.
+
    > Example
-     Compare two integers.
-         1 > 10 == False
+     Checking if 1 is greater than 10.
+         1 > 10
 Any.> : Any -> Boolean
 Any.> that = this.compare_to that == Ordering.Greater
 
+## Checks if `this` is greater than or equal to `that`.
+
+   Arguments:
+   - that: The value to compare `this` against.
+
+   To have `>=` defined, a type must define both `>` and `==`.
+
+   ! Implementing Greater Than or Equal
+     In general we recommend that you do not implement this by hand to avoid
+     semantic incorrectness. It is very rare that a more efficient
+     implementation can be provided than the one given here.
```

Comment:
Actually isn't this definition not-really that efficient for complex types?

If the types are very long strings, computing `==` and `>` can be a costly operation. This implementation can then do 2 traversals of the strings.
 Instead we could compute `compare_to` which could (theoretically, it's not currently implemented that way) do onlyone traversal of the strings and decide the order between them.

The same holds for any other type that has a complex to compute ordering, for example lexicographically ordered tuples etc.

(I am not proposing to improve the definition right now, but I think that the part about more efficient implementations is not really true; unless we decide that complex types are rare)